File: benchmarks/moonbox-live/.vendor/llmtrim/crates/llmtrim-uniffi/tests/python/test_llmtrim.py
"""End-to-end tests for the llmtrim Python bindings (the installed wheel).

Run after building + installing the wheel:
    crates/llmtrim-uniffi/scripts/build-wheel.sh
    pip install target/wheels/llmtrim-*.whl
    pytest crates/llmtrim-uniffi/tests/python

Tests use explicit presets so they don't depend on ambient LLMTRIM_* environment config.
"""

import json

import pytest

import llmtrim


def _openai(content="hello world"):
    return json.dumps(
        {"model": "gpt-4o", "messages": [{"role": "user", "content": content}], "max_tokens": 5}
    )


def test_compress_returns_projected_fields():
    out = llmtrim.compress(_openai(), llmtrim.Provider.OPEN_AI, "safe")
    assert out.provider == "openai"
    assert out.model == "gpt-4o"
    assert out.tokenizer_exact is True
    assert out.input_tokens_before > 0
    assert out.request_json  # non-empty compressed body
    json.loads(out.request_json)  # still valid JSON


def test_agent_preset_compresses_a_tool_result():
    diff = "".join(
        f"diff --git a/f{i}.rs b/f{i}.rs\n@@ -1,2 +1,2 @@\n-old {i}\n+new {i}\n" for i in range(40)
    )
    req = json.dumps(
        {
            "model": "claude-3-5-sonnet-20241022",
            "messages": [
                {"role": "user", "content": [{"type": "tool_result", "tool_use_id": "t", "content": diff}]}
            ],
            "max_tokens": 1024,
        }
    )
    out = llmtrim.compress(req, llmtrim.Provider.ANTHROPIC, "agent")
    assert out.provider == "anthropic"
    assert out.input_tokens_after < out.input_tokens_before


def test_auto_detect_provider_when_none():
    req = json.dumps({"system": "s", "messages": [{"role": "user", "content": "hi"}], "max_tokens": 5})
    out = llmtrim.compress(req, None, "safe")
    assert out.provider == "anthropic"


def test_unknown_preset_raises():
    with pytest.raises(llmtrim.LlmtrimError.UnknownPreset):
        llmtrim.compress(_openai(), llmtrim.Provider.OPEN_AI, "no-such-preset")


def test_invalid_json_raises():
    with pytest.raises(llmtrim.LlmtrimError.Compress):
        llmtrim.compress("not json", llmtrim.Provider.OPEN_AI, "safe")

File: benchmarks/moonbox-live/.vendor/llmtrim/crates/llmtrim-uniffi/tests/ruby/test_llmtrim.rb
# End-to-end tests for the llmtrim Ruby bindings.
#
# Run after generating the Ruby glue and building the cdylib:
#   crates/llmtrim-uniffi/scripts/generate-bindings.sh /tmp/gen   # emits ruby/llmtrim_ffi.rb
#   ruby -I/tmp/gen/ruby -r./llmtrim_ffi \
#     crates/llmtrim-uniffi/tests/ruby/test_llmtrim.rb
# with LD_LIBRARY_PATH (or DYLD_LIBRARY_PATH) pointing at the dir holding libllmtrim_ffi.
#
# Uses presets so it doesn't depend on ambient LLMTRIM_* environment config. Plain
# minitest (ships with Ruby) + the `ffi` gem that the generated glue requires.

require "minitest/autorun"
require "json"
require "llmtrim_ffi"

class TestLlmtrim < Minitest::Test
  def openai(content = "hello world")
    JSON.generate("model" => "gpt-4o",
                  "messages" => [{ "role" => "user", "content" => content }],
                  "max_tokens" => 5)
  end

  def test_compress_returns_projected_fields
    out = LlmtrimFfi.compress(openai, LlmtrimFfi::Provider::OPEN_AI, "safe")
    assert_equal "openai", out.provider
    assert_equal "gpt-4o", out.model
    assert out.tokenizer_exact
    assert out.input_tokens_before > 0
    refute_empty out.request_json
    JSON.parse(out.request_json) # still valid JSON
  end

  def test_agent_preset_compresses_a_tool_result
    diff = (0...40).map { |i| "diff --git a/f#{i}.rs b/f#{i}.rs\n@@ -1,2 +1,2 @@\n-old #{i}\n+new #{i}\n" }.join
    req = JSON.generate("model" => "claude-3-5-sonnet-20241022",
                        "messages" => [{ "role" => "user",
                                         "content" => [{ "type" => "tool_result", "tool_use_id" => "t", "content" => diff }] }],
                        "max_tokens" => 1024)
    out = LlmtrimFfi.compress(req, LlmtrimFfi::Provider::ANTHROPIC, "agent")
    assert_equal "anthropic", out.provider
    assert out.input_tokens_after < out.input_tokens_before
  end

  def test_auto_detect_provider_when_nil
    req = JSON.generate("system" => "s",
                        "messages" => [{ "role" => "user", "content" => "hi" }],
                        "max_tokens" => 5)
    out = LlmtrimFfi.compress(req, nil, "safe")
    assert_equal "anthropic", out.provider
  end

  def test_unknown_preset_raises
    assert_raises(LlmtrimFfi::LlmtrimError::UnknownPreset) do
      LlmtrimFfi.compress(openai, LlmtrimFfi::Provider::OPEN_AI, "no-such-preset")
    end
  end

  def test_invalid_json_raises
    assert_raises(LlmtrimFfi::LlmtrimError::Compress) do
      LlmtrimFfi.compress("not json", LlmtrimFfi::Provider::OPEN_AI, "safe")
    end
  end
end
